### PR TITLE
Formatting fixes, documentation, and a rename (note: breaking changes)

### PR DESCRIPTION
**Pull request by BytewaveMLP**
* Fit Microsoft's recommended C# naming convention by PascalCasing all method and property names.
* Give all Query classes better property names.
* Make `CONSTRAINT_CREATED` and `CONSTRAINT_UPDATED` actually useful.
* Rename the `Coolbooru` class to `CoolRequests` (**note:** needs review for a better name). Removes the necessity for `using static Coolbooru.Coolbooru` and makes code using Coolbooru look nicer.
* Add a boatload of documentation to properties of query and result objects.

### CODE REVIEW

File: Coolbooru/Coolbooru.cs [hunk 1]
```diff
@@ -18,54 +24,159 @@ public class CoolSearch {
...
 		public string sha512_hash;
+		/// <summary>
+		/// The image file's SHA512 hash before it was optimized by Derpibooru.
+		/// </summary>
 		public string orig_sha512_hash;
+		/// <summary>
+		/// The image's source URL, if provided.
+		/// </summary>
 		public string source_url;
 		public CoolRepresentation representations;
```
Comment (antigravities) [inline Coolbooru/Coolbooru.cs:127]: Should say "representations of the image," no?

File: Coolbooru/Coolbooru.cs [hunk 2]
```diff
@@ -82,39 +193,78 @@ public class CoolImages {
...
-			get { return pq; }
-			set { pq = Uri.EscapeUriString(value); }
+		/// <summary>
+		/// Your Derpibooru API key.
+		/// </summary>
+		public string APIKey;
+		/// <summary>
+		/// The Derpibooru query to search for. Defaults to *, for all tags.
+		/// </summary>
+		public string Query {
```
Comment (antigravities) [inline Coolbooru/Coolbooru.cs:209]: Should probably add something like "equivalent to 'q' on an API request." Same with all of the other things with changed names

File: Coolbooru/Coolbooru.cs [hunk 3]
```diff
@@ -157,63 +378,155 @@ public class CoolImageQuery {
...
+		/// </summary>
 		public string thumbnail_url;
 	}
 
 	/// <summary>
 	/// The main functions of Coolbooru.
 	/// </summary>
-	public class Coolbooru {
-
+	public class CoolRequests {
```
Comment (antigravities) [inline Coolbooru/Coolbooru.cs]: I don't necessarily agree with this title, it makes it sound like it's a representation of something. Maybe CoolMethods would be a better name? (this is the main thing blocking a merge)

File: Coolbooru/Coolbooru.cs [hunk 4]
```diff
@@ -157,63 +378,155 @@ public class CoolImageQuery {
...
+		/// <summary>
+		/// The ID constraint for use with List searches.
```
Comment (antigravities) [inline Coolbooru/Coolbooru.cs:491]: Lowercase "id" to make it match with the others

File: Coolbooru/Coolbooru.cs [hunk 5]
```diff
@@ -345,69 +650,93 @@ public class Coolbooru {
...
 		/// </summary>
 		/// <param name="url">A URL.</param>
 		/// <returns>A CoolEmbed object representing the result.</returns>
-		public static async Task<CoolEmbed> embed(string url) {
-			return await doResponse<CoolEmbed>("https://derpibooru.org/oembed.json?url=" + url);
+		public static async Task<CoolEmbed> Embed(string url) {
+			return await DoResponse<CoolEmbed>("https://derpibooru.org/oembed.json?url=" + url);
 		}
 	}
 }
```
Comment (antigravities) [inline Coolbooru/Coolbooru.cs:745]: Great job!

**Review by antigravities — CHANGES_REQUESTED** [review]
See comments

File: Coolbooru/Coolbooru.cs (hunk 1 above)
Comment (BytewaveMLP) [reply Coolbooru/Coolbooru.cs:127]: Ah, right, must've glossed over that thinking it was another `interactions`.

File: Coolbooru/Coolbooru.cs (hunk 2 above)
Comment (BytewaveMLP) [reply Coolbooru/Coolbooru.cs:209]: Shouldn't that be implied? It's a wrapper for a reason; though it's lower-level, we're still making a layer of abstraction on top of things.

Comment (antigravities) [reply Coolbooru/Coolbooru.cs:209]: I guess you're right

**Review by antigravities — APPROVED** [review]
:ship:

**Comment by BytewaveMLP** [comment]
👍

**Pull request merged**